src/Wrapper/GenericAKVWrapper.php

```diff
@@ -188,12 +226,39 @@ public function decrypt(string $encryptedData): string
         } catch (Throwable $e) {
             throw new ApplicationException('Deciphering failed.', $e->getCode(), $e);
         }
-        $this->verifyMetadata($decryptedContext[self::METADATA_INDEX], $this->metadata);
+        $this->verifyMetadata($decryptedContext[self::METADATA_INDEX], $metadata);
         try {
             $key = Key::loadFromAsciiSafeString($decryptedContext[self::KEY_INDEX]);
             return Crypto::decrypt($encrypted[self::PAYLOAD_INDEX], $key, true);
         } catch (Throwable $e) {
+            $doBackfill = false;
             throw new ApplicationException('Deciphering failed.', $e->getCode(), $e);
+        } finally {
+            if ($doBackfill) {
+                if (isset($this->metadata['stackId']) && self::getTransStackId()) {
```

Comment:
nevím jestli by to nemělo failnout když transStackId není nastavený

Reply:
Napadá mě, když už máme to logování, že bych tu absenci transStackId mohl lognout jako error a backfill neprovést.